cns/restserver/internalapi.go

```diff
@@ -94,32 +92,41 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	}
 	service.RUnlock()
 
-	// check if the version is valid and save it to service state
-	for ncid, nc := range ncsToBeAdded {
-		nmaReq := nmagent.NCVersionRequest{
-			AuthToken:          nc.AuthorizationToken,
-			NetworkContainerID: nc.NetworkContainerid,
-			PrimaryAddress:     nc.PrimaryInterfaceIdentifier,
-		}
-
-		ncVersionURLs.Store(nc.NetworkContainerid, nmaReq)
-		waitingForUpdate, _, _ := service.isNCWaitingForUpdate(nc.Version, nc.NetworkContainerid)
-
-		body, _ = json.Marshal(nc)
-		req, _ = http.NewRequest(http.MethodPost, "", bytes.NewBuffer(body))
-		req.Header.Set(common.ContentType, common.JsonContent)
+	skipNCVersionCheck := false
+	ctx, cancel := context.WithTimeout(context.Background(), ContextTimeOut)
+	defer cancel()
+	ncVersionListResp, err := service.nma.GetNCVersionList(ctx)
+	if err != nil {
+		skipNCVersionCheck = true
+		logger.Errorf("failed to get nc version list from nmagent")
+	}
 
-		w := httptest.NewRecorder()
-		service.createOrUpdateNetworkContainer(w, req)
+	if !skipNCVersionCheck {
```

Comment:
Doesn't look like we ever run any code if this condition doesn't match. Instead, I suggest to eliminate this variable `skipNCVersionCheck` completely and just return in the `err` check block. Then you can remove this `if` statement and it would be cleaner.